Osiris/Hacks/Resolver.cpp

```diff
@@ -0,0 +1,175 @@
+                record->wasUpdated = false;
+            }
+
+            record->prevhealth = entity->health();
+        }
+
+		for (int i = 1; i <= interfaces->engine->getMaxClients(); i++) {
+
+            auto entity = interfaces->entityList->getEntity(i);
+
+            if (!entity || entity->isDormant() || !entity->isAlive() || !entity->isOtherEnemy(localPlayer.get()) || !entity->isPlayer())
+                continue;
+
+            if (entity == localPlayer.get())
+                continue;
+
+            auto AnimState = entity->getAnimstate();
+            if (!AnimState)
+                continue;
+
+            auto activeWeapon = localPlayer->getActiveWeapon();
+            if (!activeWeapon || !activeWeapon->clip())
+                return;
+
+            auto weaponIndex = getWeaponIndex(activeWeapon->itemDefinitionIndex2());
+            if (!weaponIndex)
+                return;
+
+            if (config->ragebot[weaponIndex].resolver) {
+                Resolver::basic(entity);   
+            }
+		}
+	}
+
+	void basic(Entity* entity) noexcept
+	{
+        auto record = &records.at(entity->index());
+
+        if (!record || record->invalid)
+            return;
+
+        if (entity->velocity().length2D() > 3.0f) {
+            record->PreviousEyeAngle = entity->eyeAngles().y;
+            return;
+        }       
+
+        float desyncAng = 0;
+
+        auto animstate = entity->getAnimstate();
+
+        if (!animstate)
+            return;
+
+        if (!record->FiredUpon || !record->wasTargeted) {
+
+            entity->updateState(animstate, entity->eyeAngles());
+            if ((record->wasUpdated == false) && (entity->eyeAngles().y != record->PreviousEyeAngle)) {
+                //record->PreviousEyeAngle = entity->eyeAngles().y;
+                record->eyeAnglesOnUpdate = entity->eyeAngles().y;
+                record->prevSimTime = entity->simulationTime();
+                record->PreviousEyeAngle = entity->eyeAngles().y + record->PreviousDesyncAng;
+                record->wasUpdated = true;
+            }
+
+            /*if ((record->wasUpdated == true) && (entity->eyeAngles().y != record->PreviousEyeAngle) && (record->prevSimTime != entity->simulationTime())) {
+                record->PreviousEyeAngle = entity->eyeAngles().y;
+                
+                record->eyeAnglesOnUpdate = entity->eyeAngles().y;
+                record->PreviousEyeAngle = entity->eyeAngles().y + record->PreviousDesyncAng;
+                record->prevSimTime = entity->simulationTime();
+            }*/
+
+            entity->eyeAngles().y = record->PreviousEyeAngle;
+            entity->updateState(animstate, entity->eyeAngles());
+
+            return;
+        }
+
+        entity->updateState(animstate, entity->eyeAngles());
+
+        int missed = record->missedshots;
+        if (record->lastworkingshot != -1)
+            missed = record->lastworkingshot;
+
+        switch (missed % 9) {
```

Comment:
Where side detection? Even something basic like the following would be better
`float difference = AngleDiff ( eyeYaw, record->animstate->m_flFootYaw );
            desyncSide = 2 * difference <= 0.0f ? 1 : -1;`